src/Functions/array/arrayMap.cpp

```diff
@@ -6,7 +6,27 @@ namespace DB
 
 REGISTER_FUNCTION(ArrayMap)
 {
-    factory.registerFunction<FunctionArrayMap>();
+    FunctionDocumentation::Description description = R"(
+Returns an array obtained from the original arrays by application of `func(arr1[i], ..., arrN[i])` for each element.
+Arrays `arr1` ... `arrN` must have the same number of elements.
+
+`arrayMap` is a [higher-order function](/sql-reference/functions/overview#higher-order-functions). You must pass a lambda function to it as the first argument, and it can't be omitted.
```

Comment:
Suggest to omit l. 13. Below "FunctionDocumentation::Arguments" make it clear that the 1st argument can't be omitted.